### PR TITLE
Made Email Templates configurable.

### LINKED ISSUE #10638: Pluggable Email Template Styling
**Is your feature request related to a problem? Please describe.**
Currently, the problem is that we cannot have a different implementation depending on the use case for OSS and Collate, need to have a generic interface and concrete implementation in OSS and Collate.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by parthp2107**
### Describe your changes:

Fixes #10638 
Made Email Templates configurable.

#
### Type of change:

- [x] Improvement
- [x] New feature

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [x] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

#
### Reviewers:
@mohityadav766 @harshach

### CODE REVIEW

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EmailTemplateRepository.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    while ((readLine = bufferedReader.readLine()) != null) {
+      fileContent.append(readLine);
+    }
+    return fileContent.toString();
+  }
+
+  public void populateTemplateInDb(List<String> fileNames, String basePath) {
+    for (String fileName : fileNames) {
+      try {
+        switch (fileName) {
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EmailTemplateRepository.java:45]: why we need to have this switch statement? why not loop through and just call the storeEmailTemplate

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/email/EmailTemplateResource.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+import org.openmetadata.schema.email.EmailTemplateConfig;
+import org.openmetadata.schema.email.SmtpSettings;
+import org.openmetadata.schema.settings.SettingsType;
+import org.openmetadata.service.jdbi3.CollectionDAO;
+import org.openmetadata.service.jdbi3.EmailTemplateRepository;
+import org.openmetadata.service.resources.Collection;
+import org.openmetadata.service.resources.settings.SettingsCache;
+
+@Slf4j
+@Path("/v1/emails")
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/resources/email/EmailTemplateResource.java:36]: /system/alerts/emailTemplates

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EmailTemplateRepository.java (hunk 1 above)
Comment (parthp2107) [reply openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EmailTemplateRepository.java]: Addressed in the latest commit.

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EmailTemplateRepository.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    String readLine;
+    if (file.exists()) {
+      bufferedReader = new BufferedReader(new FileReader(basePath + fileName));
+    } else {
+      file = new File("");
+      String path = file.getAbsolutePath();
+      String[] rootPath = path.split("openmetadata-dist/");
+      bufferedReader =
+          new BufferedReader(
+              new FileReader(rootPath[0] + "/openmetadata-service/src/main/resources/emailTemplates/" + fileName));
```
Comment (mohityadav766) [inline openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EmailTemplateRepository.java:33]: this path cannot be hardcoded here

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/email/EmailTemplateResource.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    emailConfig = SettingsCache.getInstance().getSetting(SettingsType.EMAIL_CONFIGURATION, SmtpSettings.class);
+    if (emailConfig.getEmailTemplate() != null) {
+      if (!(dao.emailTemplateDAO().getEmailTypes().size() == 8)) {
+        populateTemplateInDb();
+      } else LOG.info("templates are already present");
+    }
+  }
+
+  private static void populateTemplateInDb() {
+    EmailTemplateConfig emailTemplateConfig = emailConfig.getEmailTemplate();
```
Comment (mohityadav766) [inline openmetadata-service/src/main/java/org/openmetadata/service/resources/email/EmailTemplateResource.java:58]: move this logic out of resource to some util class

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EmailTemplateRepository.java (hunk 3 above)
Comment (parthp2107) [reply openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EmailTemplateRepository.java]: Addressed in the latest commit.

File: openmetadata-service/src/main/java/org/openmetadata/service/util/EmailTemplateUtil.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import org.openmetadata.schema.email.EmailTemplateConfig;
+import org.openmetadata.schema.email.SmtpSettings;
+import org.openmetadata.service.jdbi3.EmailTemplateRepository;
+
+public class EmailTemplateUtil {
+
+  public static void populateTemplateInDb(SmtpSettings emailConfig, EmailTemplateRepository emailTemplateRepository) {
+    EmailTemplateConfig emailTemplateConfig = emailConfig.getEmailTemplate();
+    String basePath = emailTemplateConfig.getEmailTemplateBasePath();
+    Map<String, String> fileMap = new HashMap<>();
```
[the remark this answers is not in the document]
Comment (parthp2107) [reply openmetadata-service/src/main/java/org/openmetadata/service/util/EmailTemplateUtil.java:14]: It is mapping the file name with its type so we can read the content of that file and store it in the db against its type.

[the remark this answers is not in the document]
Comment (parthp2107) [reply openmetadata-service/src/main/java/org/openmetadata/service/util/EmailTemplateUtil.java:14]: Should I try and configure it like [this](https://github.com/open-metadata/OpenMetadata/blob/b715208d2825e0b3d579557079d07dc2da419e1a/openmetadata-service/src/main/java/org/openmetadata/service/elasticsearch/ElasticSearchIndexDefinition.java#L84)? @harshach

**Pull request closed without merging**